src/TraceEvent/Stacks/Linux/LinuxPerfScriptStackSource.cs

```diff
@@ -291,17 +292,27 @@ public LinuxPerfScriptStackSource(string path, bool doThreadTime)
 
         public double TotalBlockedTime { get; set; }
 
+        #region private
+        internal /*protected*/ GrowableArray<LinuxPerfScriptStackSourceSample> m_LinuxPerfScriptSamples;
+        #endregion
+
+        public LinuxPerfScriptStackSourceSample GetLinuxPerfScriptSampleByIndex(StackSourceSampleIndex sampleIndex)
```

Comment:
Is this to avoid the cast from `StackSourceSample` to `LinuxPerfScriptStackSourceSample`?

Reply:
yes, because the caller would just have to magically know that you could do that which isn't pretty. It's not super elegant API surface because the base .GetSampleByIndex() still shows up, but it seems it's required for this approach

Reply:
Ok, sounds good.